cirq-core/cirq/value/measurement_key_test.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+import cirq
+
+
+def test_empty_init():
+    with pytest.raises(TypeError, match='required positional argument'):
+        _ = cirq.MeasurementKey()
+    mkey = cirq.MeasurementKey('')
+    assert mkey.name == ''
```

Comment:
Now that we have a class, we might as well prevent empty keys - they can only generate confusion.

Reply:
Agreed. A bunch of tests failed since sometimes we don't bother naming the measurements in tests. I didn't want to clutter this PR with all that changes so will send that in a subsequent PR as soon as this is in.

Reply:
If you don't name the measurement doesn't it give it an automatic name of `str(qubits)`?

Reply:
`cirq.measure` does. But `cirq.MeasurementGate`, being a gate, does not know what qubits it's going to be operating on. So the tests which directly use the gate with `on` would fail and need explicit keys. 

As part of special treatment for qubits (phase 2b of the RFC), `MeasurementGate.on` would be overridden to pass the qubit info to `MeasurementKey` which would do remapping and maybe create the default qubit-based key, if needed.

Reply:
Thanks for the clarification!

Reply:
Sounds good to me - we can handle extra validation in subsequent PRs.